### DISCUSSION TITLE
Why there is AnchorTag in liquid?

### DISCUSSION THREAD

**Discussion by ns8482e** [OrchardCMS/OrchardCore#10422] [Q&A] [answered]
Why there is explicit [AnchorTag ](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Tags/AnchorTag.cs)in liquid ? Shouldn't the liquid tag resolve to [AnchorTagHelper](https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.TagHelpers/src/AnchorTagHelper.cs) of MVC?

The AnchorTag in liquid is actually breaking the ASP.NET AnchorTagHelper or any other tag helper that targets `a` tag

**Comment by Skrypt** [comment]
I think you need to use "all_route_data" in that case. Else the anchor gets parsed differently.

**Reply by ns8482e** [reply]
Further looking at the source code - It's broken because it's not using `TagHelper` Bridge  so for that tag is fixed to liquid tags. I feel it should revert back to use TagHelper bridge for extensibility 
// @jtkech @sebastienros

**Reply by jtkech** [reply]
@ns8482e 

This is for perf reasons

Hmm, try to use the `{% block "mvctag"` or the `{% helper "mvctag"` (for a self closed tag) liquid tags hoping it will hit the mvc tags.

**Comment by sebastienros** [comment, marked as answer]
If the liquid anchor tag is missing some feature from the Razor one, then these should be added. Don't use the bridge if it doesn't have to, it's very slow.

If there are synergies between the two worlds, then the solution should be to implement a common model, and then reuse it from liquid and razor. The current "bridge" from liquid to razor needs too much reflection, and runtime discovery.
